enterprise/utils.py

```diff
@@ -1010,3 +1022,26 @@ def discovery_query_url(content_filter):
         '<a href="{url}" target="_blank">Preview</a>',
         url=disc_url
     )
+
+
+def create_order_data_for_learner_enrollment(ecommerce_client, user, course_id):
+    """
+    Create ecommerce order data for learner and also create `CourseEnrollmentAttribute` for the learner enrollment.
+    """
+    response = ecommerce_client.create_order_for_manual_course_enrollment(
+        user.id,
+        user.username,
+        user.email,
+        course_id,
+    )
+
+    enrollment = CourseEnrollment.get_enrollment(user, course_id)
```

Comment:
@brittneyexline This line of code is giving hard time on devstack. After the successful enrollment, `CourseEnrollment.get_enrollment` always return `None`. I have tried all the possible ways to get the enrollment object but no luck. I If I put a breakpoint in the above code and check the mysql table(`student_courseenrollment`) then I can see the newly create enrollment there. I have also tried `CourseEnrollment.objects.get/filter` but no luck. I have also debugged the lms enrollment endpoint and it is working as expected and creates the enrollment but the above line fails to get that enrollment. Also if I use `EnrollmentApiClient` to get the latest enrollment than I get the enrollment but I don't want to call the api.

The weird thing is that, the above code works perfectly fine on sandbox. I have double checked the code on sandbox and it works as expected.

Any idea what is happening here? Any tip to fix the issue?

Reply:
Though I didn't figured out the issue but I think instead of directly importing the `CourseEnrollment` and `CourseEnrollmentAttribute`, It would be better to update the [enrollment api post endpoint](https://github.com/edx/edx-platform/blob/233bf245ec29f8c02cb9cf2998b84653a047a654/openedx/core/djangoapps/enrollments/views.py#L664) to handle `enrollment_attrs` updates. 
Thoughts? Any concern on this?